### ISSUE TITLE
include the node package with script tag in html

### ISSUE THREAD

**Issue by XiayidanAlimu** [focus-trap/focus-trap#100]
do anyone tell me how to include the dependence on this package on html page with script tag. 

As we can not use require() in client code, i have to add dependency that I have with script tag.

thank you in advance.

**Comment by philwolstenholme** [comment]
It would be best if the package provided a built/UMD version for use in the browser for projects without a bundler, but as a workaround you can use [bundle.run](https://github.com/Rich-Harris/packd), as long as you don't mind depending on an external service:

https://bundle.run/focus-trap@6.2.2

**Comment by stefcameron** [comment]
@philwolstenholme Thanks for the workaround, but the package _does_ provide a UMD build: `./dist/focus-trap.umd[.min].js`. I see the first inquiry by @XiayidanAlimu was posted quite a long time ago, before I took over as maintainer, and I think the UMD build was added some time last Summer (around Sep 2020, so back to about version 5.1.0). I hope that helps!

```bash
$ tree node_modules/focus-trap/
├── CHANGELOG.md
├── LICENSE
├── README.md
├── SECURITY.md
├── dist
│   ├── focus-trap.esm.js
│   ├── focus-trap.esm.js.map
│   ├── focus-trap.esm.min.js
│   ├── focus-trap.esm.min.js.map
│   ├── focus-trap.js
│   ├── focus-trap.js.map
│   ├── focus-trap.min.js
│   ├── focus-trap.min.js.map
│   ├── focus-trap.umd.js
│   ├── focus-trap.umd.js.map
│   ├── focus-trap.umd.min.js
│   └── focus-trap.umd.min.js.map
├── index.d.ts
├── index.js
└── package.json
```

**Comment by philwolstenholme** [comment]
@stefcameron ah sorry, I should have been more specific:

I'm in a similar situation, trying to use focus-trap in a CMS project with no bundler as part of the build process.

I saw the UMD files in the `dist` directory but I don't think they work as I think they still contain a `require("tabbable")` call.

We can see the difference by comparing https://cdn.jsdelivr.net/npm/focus-trap@6.2.3/dist/focus-trap.umd.min.js to https://bundle.run/focus-trap@6.2.2 .

It might be that I've missed something or am wrong, but to me it looks like the https://cdn.jsdelivr.net/npm/focus-trap@6.2.3/dist/focus-trap.umd.min.js isn't fully bundled?

**Comment by philwolstenholme** [comment]
(Also I hadn't spotted that the original ticket was from 2019, oops!)

**Comment by stefcameron** [comment]
@philwolstenholme I see. That's right: We don't bundle `tabbable` into `focus-trap`. Both libraries come with UMD builds now. To use `focus-trap` as UMD, you would need to also include a `<script src="/path/to/tabbable.umd.min.js"></script>` tag, before the one that loads `focus-trap`.

In a UMD scenario, in the browser (where CJS and AMD are not used), `focus-trap` should be getting loaded with line 11 of the `focus-trap.umd.js` (non-minified) file:

```javascript
factory(exports, global.tabbable);
```

In other words, because CJS and AMD aren't found, the bundle resorts to expecting that `global.tabbable` is defined, which means `window.tabbable`, and the result of loading the `tabbable.umd.js` prior should be that `window.tabbable` gets defined.

So it should all work.

FYI, you prompted me to check over the README and I just updated it to try to clarify how to use the [UMD build](https://github.com/focus-trap/focus-trap/blob/master/README.md#umd).

**Comment by philwolstenholme** [comment]
@stefcameron Oh, I see! Yes, I've just tested it and that works - thank you very much for the explanation. I'd assumed it was all self-contained but the approach makes sense now. 

Thanks for updating the README too, that will be really useful for a few people in the future I imagine 👍

**Comment by stefcameron** [comment]
@philwolstenholme Awesome! 🎉
